tests/pytests/functional/modules/test_mac_group.py

```diff
@@ -0,0 +1,193 @@
+"""
+    :codeauthor: Nicole Thomas <nicole@saltstack.com>
+"""
+
+import pytest
+from saltfactories.utils import random_string
+
+pytestmark = [
+    pytest.mark.slow_test,
+    pytest.mark.destructive_test,
+    pytest.mark.skip_if_not_root,
+    pytest.mark.skip_unless_on_darwin,
+]
+
+
+@pytest.fixture(scope="module")
+def group(modules):
+    return modules.group
+
+
+# Create group name strings for tests
+@pytest.fixture(scope="module")
+def add_group():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module")
+def del_group():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module")
+def change_group():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module")
+def add_user():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module")
+def rep_user_group():
+    yield random_string("RS-", lowercase=False)
+
+
+@pytest.fixture(scope="module", autouse=True)
+def _setup_teardown_vars(group, add_group, change_group, del_group):
+    try:
+        yield
+    finally:
+        # Delete ADD_GROUP
+        add_info = group.info(add_group)
+        if add_info:
+            group.delete(add_group)
+
+        # Delete DEL_GROUP if something failed
+        del_info = group.info(del_group)
+        if del_info:
+            group.delete(del_group)
+
+        # Delete CHANGE_GROUP
+        change_info = group.info(change_group)
+        if change_info:
+            group.delete(change_group)
+
+
+def test_mac_group_add(group, add_group):
+    """
+    Tests the add group function
+    """
+    group.add(add_group, 3456)
+    group_info = group.info(add_group)
+    assert group_info["name"] == add_group
+
+    group.delete(add_group)
+
+
+def test_mac_group_delete(group, del_group):
+    """
+    Tests the delete group function
+    """
+    # Create a group to delete - If unsuccessful, skip the test
+    group_add_ret = group.add(del_group, 4567)
+    if group_add_ret is not True:
+        group.delete(del_group)
+        pytest.skip("Failed to create a group to delete")
+
+    # Now try to delete the added group
+    ret = group.delete(del_group)
+    assert ret
+
+
+def test_mac_group_chgid(group, change_group):
+    """
+    Tests changing the group id
+    """
+    # Create a group to delete - If unsuccessful, skip the test
+    ret = group.add(change_group, 5678)
+    if ret is not True:
+        group.delete(change_group)
+        pytest.skip("Failed to create a group to manipulate")
+
+    group.chgid(change_group, 6789)
+    group_info = group.info(change_group)
+    assert group_info["gid"] == 6789
+
+    group.delete(change_group)
```

Comment:
```suggestion
```

Reply:
I needed this in there for the tests to pass, otherwise the group was when the next test ran.  The `finally` was only running after all the tests had run and doing the delete.

Reply:
I added a few more changes to fix that.
The way it is now, without my proposed changes, if a test fails, after the group was created, then the next test will have an existing group, which is probably not what we need.